OpenRA.Game/Map/MapCache.cs

```diff
@@ -378,4 +401,118 @@ public void Dispose()
 			});
 		}
 	}
+
+	public class MapDirectoryTracker
+	{
+		readonly FileSystemWatcher watcher;
+		readonly ModData modData;
+		readonly MapGrid mapGrid;
+		readonly IReadOnlyPackage package;
+		readonly MapClassification classification;
+
+		enum MapAction { Add, Delete, Update }
+		readonly Dictionary<string, MapAction> mapActionQueue = new Dictionary<string, MapAction>();
```

Comment:
I'm happy to change this queue mechanism to whatever for is standard in ora engine. Just point me towards an example

Reply:
The data type can be whatever collection is most appropriate for your use case. Your current selection seems fine to me.

Reply:
Great